README.md

```diff
@@ -123,6 +123,26 @@ let x = 0.0
 end
 ```
 
+Note that the let block is used here to create a scope to allow assignment of 
+`x` within `@byrow!`.
+
+`byrow!` also supports special syntax for allocating new columns to make
+`byrow!` more useful for data transformations. The syntax `@newcol
+x::Array{Int}` allocates a new column `:x` with an `Array` container with eltype
+`Int`. Note that the returned AbstractDataFrame includes these new columns, but
```

Comment:
@andyferris As you can see above, the DataFramesMeta API should work fine for TypedTables too. If you find cases where you'd need a different API, filing an issue to discuss this could be a good idea.